integration/bdpl_update_test.go

```diff
@@ -348,20 +348,27 @@ var _ = Describe("BDPL updates", func() {
 
 			It("should update when the user's secret changes", func() {
 				err := env.WaitForSecret(env.Namespace, "desired-manifest-v2")
-				Expect(err).NotTo(HaveOccurred(), "error waiting for new desired manifest")
+				Expect(err).NotTo(HaveOccurred(), "error waiting for new desired manifest v2")
 
 				_, tearDown, err := env.UpdateSecret(env.Namespace, env.UserExplicitPassword("my-var", "anothersupersecret"))
 				Expect(err).NotTo(HaveOccurred(), "error updating user var")
 				tearDowns = append(tearDowns, tearDown)
 
 				err = env.WaitForSecret(env.Namespace, "desired-manifest-v3")
-				Expect(err).NotTo(HaveOccurred(), "error waiting for new desired manifest")
+				Expect(err).NotTo(HaveOccurred(), "error waiting for new desired manifest v3")
 
-				secret, err := env.GetSecret(env.Namespace, "desired-manifest-v3")
-				Expect(err).NotTo(HaveOccurred(), "error getting new desired manifest")
+				err = env.CustomWaitForSecret(env.Namespace, "desired-manifest-v4", 20*time.Second)
```

Comment:
Could we get the latest versioned secret for this case?

Reply:
Yeah. Makes sense. Let me update that.

Reply:
Removed that change. Will make another PR is necessary.